GameServer.cpp

```diff
@@ -144,18 +143,16 @@ int GameServer::AllocSocket(SOCKET& hostSocket, unsigned short port)
 	int errorCode = bind(hostSocket, (sockaddr*)&hostAddress, sizeof(hostAddress));
 
 	if (errorCode == SOCKET_ERROR) {
-		return SocketBindFailed;
+		throw std::runtime_error("Unable to bind socket");
 	}
 
 	// Set non-blocking mode
 	unsigned long argp = true;
 	errorCode = ioctlsocket(hostSocket, FIONBIO, &argp);
 
 	if (errorCode != 0) {
-		return SocketNonBlockingModeFailed;
+		throw std::runtime_error("Unable to set non-blocking mode on socket");

```

Comment:
I would recommend adding a formatted address to the error message (which includes the port).